File: go/sort-by-height/sort-by-height.go
package main

import (
	"fmt"
	"slices"
	
)

func sortByHeight(a []int)[]int{
	slicePerson := []int{}
	for _, val := range a{
		if val != -1{
			slicePerson = append(slicePerson, val)
		}
	}
	slices.Sort(slicePerson)

	for idx, val := range a{
		if val !=-1{
			a[idx] = slicePerson[0] 
			slicePerson = slicePerson[1:] //the two lines are like a shift in js
		}
	}
	return a
}

func main() {
	a := []int{-1, 150, 190, 170, -1, -1, 160, 180}
	fmt.Printf("The answer is: %v", sortByHeight(a))
}

File: js/sort-by-height.js
function sortByHeight(a){
    let arrayPerson = []

    for(i = 0;i < a.length; i++){
        if(a[i] != -1){
            arrayPerson.push(a[i])
        }
    }
    arrayPerson.sort((a,b) => a - b)

    for(let i = 0; i < a.length; i++) {
        if(a[i] != -1) {
            a[i] = arrayPerson.shift()
        }
    }
    
    return a
}

let a = [-1, 150, 190, 170, -1, -1, 160, 180]
console.log(sortByHeight(a))